ext/TrixiNLsolveExt.jl

```diff
@@ -0,0 +1,77 @@
+# Package extension for adding Convex-based features to Trixi.jl
+module TrixiNLsolveExt
+
+# Required for coefficient optimization in P-ERK scheme integrators
+if isdefined(Base, :get_extension)
+    using NLsolve: nlsolve
+else
+    # Until Julia v1.9 is the minimum required version for Trixi.jl, we still support Requires.jl
+    using ..NLsolve: nlsolve
+end
+
+# We use a random initialization of the nonlinear solver.
+# To make the tests reproducible, we need to seed the RNG
+using Random: seed!
+
+# Use functions that are to be extended and additional symbols that are not exported
+using Trixi: Trixi, solve_a_unknown!, PairedExplicitRK3_butcher_tableau_objective_function,
+             @muladd
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+# Find the values of the a_{i, i-1} in the Butcher tableau matrix A by solving a system of
+# non-linear equations that arise from the relation of the stability polynomial to the Butcher tableau.
+# For details, see Proposition 3.2, Equation (3.3) from 
+# Hairer, Wanner: Solving Ordinary Differential Equations 2
+function Trixi.solve_a_unknown!(a_unknown, num_stages, monomial_coeffs, c_s2, c;
+                                verbose, max_iter = 100000)
+    is_sol_valid = false
+
+    # Define the objective_function
+    function objective_function(x)
+        return Trixi.PairedExplicitRK3_butcher_tableau_objective_function(x, num_stages,
+                                                                          num_stages,
+                                                                          monomial_coeffs,
+                                                                          c_s2)
+    end
+
+    seed!(5555)
```

Comment:
Could you add a comment explaining why we need the seed (test values)? This is clear to me, but maybe not to someone looking at this in the future.